oracle/oracle.go

```diff
@@ -33,6 +35,10 @@ func (or *Oracle) AdvanceStateToNextSlot(
 	blockUnsubs []Unsubscription,
 	blockDonations []Donation) (uint64, error) {
 
+	//declare mutex and release it when function returns
+	or.mu.Lock()
```

Comment:
while this helps, it doesnt fully solve the issue. The mutex has to be actually used. An option could be to use a read mutex `or.mu.RLock()` in `main.go` where the SIGINT is captured.

If the oracle is interrupted and the mutex is locked by `AdvanceStateToNextSlot`, `or.mu.RLock()` will block the execution until its release at the end. And right after its released, the state will be stored, and the program will exit. 

see https://github.com/dappnode/mev-sp-oracle/pull/99